cmd/operator-opamp-bridge/agent/agent.go

```diff
@@ -291,7 +291,7 @@ func (agent *Agent) getEffectiveConfig(ctx context.Context) (*protobufs.Effectiv
 	}
 	instanceMap := map[string]*protobufs.AgentConfigFile{}
 	for _, instance := range instances {
-		marshaled, err := yaml.Marshal(instance)
+		marshaled, err := yaml.Marshal(&instance)
```

Comment:
From @jmacd 
> when you put a value into an interface, it becomes something you can't address, but if you put a pointer in, you can still address it. The `allowAddr` variable changes in the expression above. you can't mutate values in interfaces either, because they are not addressable